pkg/apis/cloudcore/v1alpha1/types.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	metaconfig "github.com/kubeedge/kubeedge/pkg/apis/meta/v1alpha1"
+)
+
+type CloudCoreConfig struct {
+	metav1.TypeMeta
+	// KubeAPIConfig set the kubernetes cluster info which cloudcore will connected
+	// +Required
+	KubeAPIConfig KubeAPIConfig `json:"kubeAPIConfig,omitempty"`
+	// Modules set cloudcore modules config
+	// +Required
+	Modules Modules `json:"modules,omitempty"`
+}
+
+type KubeAPIConfig struct {
+	// Master set the address of the Kubernetes API server (overrides any value in Kubeconfig)
+	// such as https://127.0.0.1:8443
+	// default ""
+	// Note: Can not use "omitempty" option,  It will affect the output of the default configuration file
+	Master string `json:"master"`
+	// ContentType set the ContentType of message transmission when interacting with k8s
+	// default application/vnd.kubernetes.protobuf
+	ContentType string `json:"contentType,omitempty"`
+	// QPS to use while talking with kubernetes apiserve
+	// default 100
+	QPS int32 `json:"qps,omitempty"`
+	// Burst to use while talking with kubernetes apiserver
+	// default 200
+	Burst int32 `json:"burst,omitempty"`
+	// Kubeconfig set path to kubeconfig file with authorization and master location information.
+	// default "/root/.kube/config"
+	KubeConfig string `json:"kubeConfig,omitempty"`
+}
+
+type Modules struct {
+	// CloudHub set cloudhub module config
+	CloudHub CloudHub `json:"cloudhub"`
+	// EdgeController set edgecontroller module config
+	EdgeController EdgeController `json:"edgecontroller"`
+	// DeviceController set devicecontroller module config
+	DeviceController DeviceController `json:"devicecontroller"`
+}
+
+type CloudHub struct {
+	// Enable indicates whether cloudhub is enabled, if set to false (for debugging etc.), skip checking other cloudhub configs.
+	// default true
+	Enable bool `json:"enable"`
```

Comment:
Enable is not required explicitly set to true in common config, so you can add `omitempty` here
```suggestion
	Enable bool `json:"enable,omitempty"`
```

Reply:
done